code/modules/mob/mob_helpers.dm

```diff
@@ -346,6 +346,29 @@ proc/slur(phrase)
 		newphrase+="[newletter]";counter-=1
 	return newphrase
 
+proc/roboslur(phrase) //for robutts
+	phrase = html_decode(phrase)
+	var/leng=lentext(phrase)
+	var/counter=lentext(phrase)
+	var/newphrase=""
+	var/newletter=""
+	while(counter>=1)
+		newletter=copytext(phrase,(leng-counter)+1,(leng-counter)+2)
+		if(rand(1,3)==3)
+			if(lowertext(newletter)=="o")	newletter="u"
+			if(lowertext(newletter)=="s")	newletter="ch"
+			if(lowertext(newletter)=="a")	newletter="ah"
+			if(lowertext(newletter)=="c")	newletter="k"
+		switch(rand(1,15))
+			if(1,3,5,)	newletter="[lowertext(newletter)]"
+			if(2,4,6,)	newletter="[uppertext(newletter)]"
+			if(7, 8, 15)	newletter+=pick("@", "!", "#", "$", "%", "&", "?")
+			//if(9,10)	newletter="<b>[newletter]</b>"
+			//if(11,12)	newletter="<big>[newletter]</big>"
+			//if(13)	newletter="<small>[newletter]</small>"
+		newphrase+="[newletter]";counter-=1

```

Comment:
oh, i see how i missed `counter -= 1`
please no semicolons, move this to a new line